lib/rule-helpers.js

```diff
@@ -384,3 +384,90 @@ module.exports.noNodeEnvInSSR = function noNodeEnvInSSR(context) {
         },
     };
 };
+
+module.exports.noUnguardedAsyncSSR = function noUnguardedAsyncSSR(context) {
+    const {
+        withinLWCVisitors,
+        isInsideReachableMethod,
+        isInsideReachableFunction,
+        isInsideSkippedBlock,
+    } = reachableDuringSSRPartial();
+
+    const asyncOperations = new Set(['fetch', 'then', 'catch', 'finally']);
```

Comment:
`try` blocks also use `catch` and `finally`. Will this rule get triggered for this?

```
try {
    // do stuff
} catch(e) {
    // handle error
} finally {
    // cleanup
}

Reply:
No it wont flag then I will also add a test case for this.